### DISCUSSION TITLE
Is it possible to have an async blocker with unstable_useBlocker

### DISCUSSION THREAD

**Discussion by lionep** [remix-run/react-router#9907] [Q&A] [answered]
Hey there!

I installed the v6.7.0-pre.3 and use the gist detailed in this [comment](https://github.com/remix-run/react-router/issues/8139#issuecomment-1382428200) to implement usePrompt.

I was wondering if I can replace the window.confirm from the useBlocker with an async function, because I already have a modal system in my app.

I tried to figure in the [hook code](https://github.com/remix-run/react-router/blob/2cd8266765925f8e4651d7caf42ebe60ec8e163a/packages/react-router/lib/hooks.tsx#L834-L852), but this does not seem possible.

Is there any reason that it can't work with async function ? (Maybe usage of before unload or something like this)

Thanks !

**Comment by machour** [comment, marked as answer]
This is not possible, see Chance's comment here: https://github.com/remix-run/react-router/pull/9709#discussion_r1045151429

**Reply by lionep** [reply]
Thanks ! I'll find another way to manage this.
